src/main/java/Game.java

```diff
@@ -0,0 +1,160 @@
+import java.util.Arrays;
+import stdin.CharSource;
+import stdout.Answer;
+import stdout.Gallows;
+
+
+/**
+ * The class of the game in which the logic of the game Gallows is located
+ */
+public class Game {
+
+  /**
+   * The field containing the hidden word
+   */
+  final private String word;
+
+  /**
+   * User's input source
+   */
+  private CharSource source;
+
+  /**
+   * Attempts user
+   */
+  private AttemptsHangman attemptsHangman;
+
+  /**
+   * The mask of the hidden word in the form of "____"
+   */
+  private StringBuilder maskWord;
+
+  /**
+   * The answer is for the user
+   */
+  private Answer answer;
+
+  /**
+   * The gallows condition field
+   */
+  private Gallows gallow;
+
+  private ArrayList<Character> playerCharList = new ArrayList<>();
+
+  /**
+   * Creates a new Game, taking into account the {@link CharSource}, attempts
+   * {@link AttemptsHangman}, the hidden {@link Game#word}, the {@link Answer} for the user
+   */
+  public Game(CharSource source, AttemptsHangman attemptsHangman, String word, Answer answer) {
+    this.source = source;
+    this.attemptsHangman = attemptsHangman;
+    this.word = word;
+    this.answer = answer;
+    this.maskWord = new StringBuilder("_".repeat(word.length()));
+    this.gallow = new Gallows();
+  }
+
+  /**
+   * The play method contains the logic of the game
+   *
+   * @return Returns the result of the game in the form of truth or false
+   */
+  public boolean play() {
+    while (this.attemptsHangman.hasDied() && !this.isMatch()) {
+
+      //Message Assembly
+      String message = this.gallow.getGallow(this.attemptsHangman.getAttempts())
+          + "\nВы использоватли следующие буквы: " + Arrays.toString(this.playerCharList.toArray())
+          + "\nНа данный момент слово выглядит так: " + this.maskWord
+          + "\nВведите свое предположение:";
+
+      this.answer.print(message);
+
+      var userAnswer = nextChar();
+      if (!checkDuplicate(userAnswer)) {
+        continue;
+      }
+
+      if (!checkCharWord(userAnswer)) {
+
+        this.attemptsHangman.registerAttempts();
+        this.playerCharList.add(userAnswer);
+        this.answer.print("Извините, такой буквы в слове нету");
+
+      } else {
+
+        this.answer.print("Да такая буква имеется ");
+        this.playerCharList.add(userAnswer);
+
+      }
+    }
+
+    return this.isMatch();
+  }
+
+  /**
+   * The method of checking the correspondence of the mask-word to the hidden word
+   *
+   * @return Returns true if the word is matched, false otherwise
+   */
+  private boolean isMatch() {
+    return this.maskWord.toString().equals(this.word);
+  }
+
+  /**
+   * The method of checking a character in a word
+   *
+   * @param ch Проверяемый символ
+   * @return Returns true if the character is in the word, false otherwise
+   */
+  private boolean checkCharWord(char ch) {
+
+    var index = 0;
+    boolean flag = false;
+    //Нахождение символа в слове и измениние маски
```

Comment:
Стоит всегда ставить пробел после косой черты
```suggestion
    // Нахождение символа в слове и измениние маски
```